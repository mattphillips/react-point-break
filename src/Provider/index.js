import React from 'react';
import { element } from 'prop-types';

import { Provider } from '../Context';

const BreakPointProvider = ({ children, ...queries }) => {
  return <Provider value={queries}>{children}</Provider>;
};

BreakPointProvider.propTypes = {
  children: element.isRequired
};

export default BreakPointProvider;
