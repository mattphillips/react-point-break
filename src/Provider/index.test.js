import React from 'react';
import { shallow } from 'enzyme';

import Provider from './';

const queries = {
  md: '(min-width: 801px) and (max-width: 900px)',
  lg: '(min-width: 901px)'
};

describe('<Provider />', () => {
  it('renders given children', () => {
    const component = shallow(
      <Provider>
        <h1>Hello, world!</h1>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders provider with given queries', () => {
    const component = shallow(
      <Provider {...queries}>
        <h1>Hello, world!</h1>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
