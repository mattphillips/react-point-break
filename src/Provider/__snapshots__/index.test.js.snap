// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Provider /> renders given children 1`] = `
<Provider
  value={Object {}}
>
  <h1>
    Hello, world!
  </h1>
</Provider>
`;

exports[`<Provider /> renders provider with given queries 1`] = `
<Provider
  value={
    Object {
      "lg": "(min-width: 901px)",
      "md": "(min-width: 801px) and (max-width: 900px)",
    }
  }
>
  <h1>
    Hello, world!
  </h1>
</Provider>
`;
